### PR TITLE
🐛 Postgres Destination fails when the connection is persisted across steps 

### PR DESCRIPTION
**Pull request by kikoncuo**
## What
https://github.com/airbytehq/airbyte/issues/39364
When using PostgreSQL as a destination, we tried to recreate a postgreSQL function which leads to an SQL error which crashes the sync process.

## How
We've modified the SQL query to either create or replace the function, not only create which would lead to an error if the function already existed

## Review guide
Try to use PostgreSQL as a destination with any source

## User Impact
The function may re-created the function when it's not needed, but this shouldn't happen anyway as it's only on the session, in case other errors happen related to re-using the session, this would not crash the connector

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by marcosmarxm** [comment]
@kikoncuo please sign the CLA and update the title and description to match guidelines.

File: airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/PostgresDestination.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ protected DataSourceFactory.DataSourceBuilder modifyDataSourceBuilder(final Data
     // Adding 60 seconds to connection timeout, for ssl connections, default 10 seconds is not enough
     return builder.withConnectionTimeout(Duration.ofSeconds(60))
         .withConnectionInitSql("""
-                               CREATE FUNCTION pg_temp.airbyte_safe_cast(_in text, INOUT _out ANYELEMENT)
+                               CREATE OR REPLACE FUNCTION pg_temp.airbyte_safe_cast(_in text, INOUT _out ANYELEMENT)
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/PostgresDestination.java:76]: You need to bump the metadata and doc chagelog version too.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by kikoncuo** [comment]
> @kikoncuo please sign the CLA and update the title and description to match guidelines.

Done!

File: airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/PostgresDestination.java (hunk 1 above)
Comment (kikoncuo) [reply airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/PostgresDestination.java:76]: You mean I should update this file:
https://github.com/airbytehq/airbyte/blob/master/.bumpversion.cfg
as a patch (last number) based on (https://semver.org/)

Or this file?
https://github.com/airbytehq/airbyte/blob/master/docs/integrations/destinations/postgres.md

Comment (kikoncuo) [reply airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/PostgresDestination.java:76]: Gonna follow the files modified in this file and manually update them
https://github.com/airbytehq/airbyte/pull/37660/files

**Comment by evantahler** [comment]
@kikoncuo - just curious, which version/pass offering of postgres are you using that you saw this bug?

**Comment by kikoncuo** [comment]
@evantahler I'm using a new default supabase project which uses Postgres version 15.1.1.58
image

I also tried with an older project from a month ago which I successfully used to do multiple syncs, same problem.
I closed this PR because it was a bit of a mess and opened a new one here: https://github.com/airbytehq/airbyte/issues/39378

**Pull request closed without merging**